src/admin_dept_management.c

```diff
@@ -0,0 +1,263 @@
+    x=0;
+    
+	do{
+    printf("Enter Phone number\n");
+    scanf("%s",emp.phone_number);
+    x=valid_phone(emp.phone_number);
+    } while(x==0);
+    x=0;
+	
+	do{
+    printf("Enter the date of joining\n");
+    printf("   Enter day of joining\n");
+    scanf("%d",&emp.day);
+    printf("   Enter month of joining\n");
+    scanf("%d",&emp.month);
+    printf("   Enter year of joining\n");
+    scanf("%d",&emp.year);
+    x=datevalid(emp.day,emp.month,emp.year);
+     } while(x==0);
+	 
+	char stmt[1500];		
+	char qry_retrieve[]={"select department,designation from employee_request where request_id='%d'"};
+	conn6=mysql_init(NULL);
+	mysql_real_connect(conn6, "localhost", "root", "1234","payroll", 3306, NULL, 0);
+	
+	if(conn6){
+		int n=sprintf(stmt,qry_retrieve,request_id);
+		mysql_query(conn6,stmt);
+		read = mysql_store_result(conn6);
+		if (mysql_query(conn6,stmt)){
+            printf(" Error: %s\n", mysql_error(conn6));
+            printf("Failed to execute query.");
+        }
+        else{
+			row = mysql_fetch_row(read);
+			strcpy(emp.dep,row[0]);
+			strcpy(emp.des,row[1]);
+        }		
+	}
+	
+	else{
+    	printf("not connected");
+        printf("%s\n", mysql_error(conn6));
+	}
+	conn7=mysql_init(NULL);
+	mysql_real_connect(conn7, "localhost", "root", "1234","payroll", 3306, NULL, 0);
+	char qry_add_emp[]={"insert into emp_details (name1,name2,department,designation,age,address_l1,address_l2,phonen,gender,email,day,month,year,emp_type) VALUES('%s','%s','%s','%s','%d','%s','%s','%s','%s','%s','%d','%d','%d','%s')"};
+
+	if(conn7){
+    	int n=sprintf(stmt,qry_add_emp,emp.first_name,emp.last_name,emp.dep,emp.des,emp.age,emp.address_line1,emp.address_line2,emp.phone_number,emp.gender,emp.email,emp.day,emp.month,emp.year,emp.status);
+        if (mysql_query(conn7,stmt)){
+   		    printf(" Error: %s\n", mysql_error(conn6));
+   		    return "Failed to execute query.";	
+ 	    }
+       	else{
+ 		   printf( "\n\n\nuser details added successfully\n\n\n\n\n");
+	    }
+    }
+    else{
+    	printf("not connected");
+        printf("%s\n", mysql_error(conn6));	
+	}
+	int emp_id;
+	char qry_get_emp_id[]={"select max(emp_id) as id from emp_details"};
+	if(conn7){
+    	int n=sprintf(stmt,qry_get_emp_id);
+        mysql_query(conn7,stmt);
+		read = mysql_store_result(conn7);
+        if (mysql_query(conn7,stmt)){
+            printf(" Error: %s\n", mysql_error(conn7));
+            printf("Failed to execute query.");
+        }
+        else{
+            int i=0;
+			row = mysql_fetch_row(read);
+            emp_id=atoi(row[0]);
+            mysql_free_result(read);        
+        }
+	}
+    else{
+    	printf("not connected");
+        printf("%s\n", mysql_error(conn7));	
+	}
+	oo2=mysql_init(NULL);
+	mysql_real_connect(oo2, "localhost", "root", "1234","payroll", 3306, NULL, 0);
+	char password []="1234";
+	char employee_type[30];
+	printf("Enter the employee type -Employee or Manager\n");
+	scanf("%s",employee_type);
```

Comment:
You can add a check here to validate employee type. I can add any employee type like with "abc"